src/jit/optcse.cpp

```diff
@@ -2348,10 +2371,32 @@ class CSE_Heuristic
 
                 noway_assert(asg->gtOp.gtOp1->gtOper == GT_LCL_VAR);
 
+                // Backpatch the SSA def, if we're putting this cse temp into ssa.
+                asg->gtOp.gtOp1->AsLclVarCommon()->SetSsaNum(cseSsaNum);
+
+                if (cseInSSA)
+                {
+                    // We should have allocated an ssa number above.
+                    assert(cseSsaNum != SsaConfig::RESERVED_SSA_NUM);
```

Comment:
I'd remove `cseInSSA` and simply test if cseSsaNum is reserved num or not. The bool is basically redundant information.